modules/promotions/promotion-data.php

```diff
@@ -0,0 +1,54 @@
+<?php
+namespace Elementor\Modules\Promotions;
+
+use Elementor\Core\Utils\Promotions\Filtered_Promotions_Manager;
+use Elementor\Includes\EditorAssetsAPI;
+use Elementor\Utils;
+
+class PromotionData {
+	protected EditorAssetsAPI $editorAssetsAPI;
+
+	public function __construct( EditorAssetsAPI $editorAssetsAPI ) {
+		$this->editorAssetsAPI = $editorAssetsAPI;
+	}
+
+	public function get_promotion_data( $force_request = false ): array {
+		$assets_data = $this->transform_assets_data( $force_request );
+
+		return [
+			Utils::ANIMATED_HEADLINE => $this->get_animated_headline_data( $assets_data ),
+		];
+	}
+
+	private function transform_assets_data( $force_request = false ) {
+		$assets_data = $this->editorAssetsAPI->get_assets_data( $force_request );
+		$transformed_data = [];
+
+		foreach ( $assets_data as $asset ) {
+			$transformed_data[ $asset['id'] ] = $asset['imageSrc'];
+		}
+
+		return $transformed_data;
+	}
+
+	private function get_animated_headline_data( $assets_data ) {
+		$data = [
+			'image' => esc_url( $assets_data[ Utils::ANIMATED_HEADLINE ] ) ?? '',
```

Comment:
@hein-obox  `esc_url( .... ) ?? ''` is this correct?
or did we planned to do `esc_url( $src[ $key ] ?? '' )` ?

Reply:
done...